CIP-1694/README.md

```diff
@@ -0,0 +1,1124 @@
+> voting.
+
+#### Registered DReps
+
+In Voltaire, existing stake credentials will also be
+able to delegate their stake to registered DReps.
+DRep delegation will mimic the existing stake delegation mechanisms (via on-chain certificates).
+Similarly, DRep registration will mimic the existing stake registration mechanisms.
+
+Registered DReps are identified by a credential that can be either:
+
+* A verification key (Ed25519)
+* A native or Plutus script
+
+The blake2b-224 hash digest of a serialized DRep credential is called the _DRep ID_.
+
+The following new types of certificates will be added for governance:
+
+##### DRep registration certificates
+
+DRep registration certificates include:
+
+* a DRep ID
+* an anchor
+
+An **anchor** is a pair of:
+
+* a URL to a JSON payload of metadata
+* a hash of the contents of this metadata URL
+
+The structure and format of this metadata is deliberately left open in this CIP.
+The on-chain rules will not check either the URL or the hash.
+Client applications should, however, perform the usual sanity checks when fetching content from the provided URL.
+
+
+##### DRep retirement certificates
+
+DRep retirement certificates include:
+
+* a DRep ID
+* the epoch number after which the DRep will retire
+
+##### Vote delegation certificates
+
+Vote delegation certificates include:
+
+* the DRep ID to which the stake should be delegated
+* the stake credential for the delegator
+
+The authorization scheme (i.e. which signatures are required for registration, retirement or delegation) mimics the existing stake delegation certificate authorization scheme.
+
+<!-- TODO: Provide CBOR specification in the annexe for those new certificates. -->
+
+
+#### New stake distribution for DReps
+
+In addition to the existing per-stake-credential distribution and the
+per-stake-pool distribution, the ledger will now also determine the per-DRep stake
+distribution. This distribution will determine how much stake is backed by each
+`Yes` vote from a DRep.
+
+> **Warning**
+>
+> **Unlike** the distribution that is used for block production, we will always use the most
+> current version of the per-DRep stake distribution as given on the epoch boundary.
+>
+> This means that **for any topic which individual voters cares deeply about,
+> they have time to register themselves as a DRep and vote directly**.
+> However, it means that there may be a difference between the stake that is used for block
+> production and the stake that is used for voting in any given epoch.
+
+#### Incentives for Ada holders to delegate voting stake
+
+After the [bootstrapping phase](#bootstrapping-phase), reward accounts will be
+**blocked from withdrawing any rewards** unless their associated stake credential is also delegated to a DRep
+(either pre-defined or registered).  This helps to ensure high participation, and so, legitimacy.
```

Comment:
For all intents and purposes, we can consider all "staked" ADA to already be actively participating in governance, indirectly, via their delegation to their SPO (who already has some voting power within the governance system). In my personal opinion **requiring** this explicit registration, particularly at the point of a holder needing/wanting to withdraw their rewards, creates a very negative sentiment towards participation in the governance system and create a lot of bandwidth for "abstain" delegations that have no functional impact on the actual outcome.

Thus, the measurable participation in governance is most likely unaffected by this change and, because participation is coerced via this negative feedback loop, I see no evidence that it will increase "legitimacy" of the vote outcome or encourage proactive, positive participation in the governance system.

Reply:
> we can consider all "staked" ADA to already be actively participating in governance

All staked ADA is only relevant to consensus, not governance. I theorise that one can partake in governance without staking to a pool. Two different systems.

> In my personal opinion requiring this explicit registration, particularly at the point of a holder needing/wanting to withdraw their rewards.

I see it as a plausible requirement by controlling engine of the ledger, i.e. Voltaire governance.

> the measurable participation in governance is most likely unaffected by this change and, because participation is coerced via this negative feedback loop

I daresay that the reasoning that provided in the CIP that is "ensure high participation, and so, legitimacy." is lazy writing to cover up a more technical constraint of the ledger. Happy to have folks such as @JaredCorduan correct me on this, it is my active theory after all and open to clarification.

Reply:
There's no technical constraint to this, it's an easy thing to drop. I fully agree that it's very unlikely that this would improve actual turnout and is mostly an attempt to improve perceived legitimacy (which I'm not sure it does). Documents like this are all about making compromises.

Reply:
I believe this places an undue burden on "becoming an active participant" at the wrong times during a user's "Cardano Journey".

1. I am forced/strong-armed at the point of creating a wallet to: Remember a seed phrase, choose from a list of 3k+ stake pools, choose from a list of 2...(n) vReps for my governance actions. Bearing in mind I'm not already an "expert" on how Cardano works and have never heard of stake pools or governance before, is this the right time to throw me in the deep end from a mass adoption/onboarding perspective?
2. I have a need (or desire) to withdraw my rewards (maybe I'm trying to cash out to a CEX to **leave** the ecosystem, purchase an NFT, do something wrt Defi, etc)... I can't withdraw my rewards until I choose from 2...(n) vReps for my governance actions. How likely am I to do due diligence and make a conscious, active, and informed choice at this point?

Strictly my opinion, but I feel coordinated "Rock the Vote" campaigns (off-chain, via social media, dApps, wallets) would be far more beneficial to encouraging positive participation "on the user's terms" than this proposed solution.

Reply:
Consider the huge difference between:

* decisions made without my (as an "Ada holder") consent or knowledge
* me explicitly opting out of having a voice

Reply:
If I am **forced** to give my consent or otherwise signal my consent or knowledge, that would be considered an action taken under **duress** (e.g. you held my staking rewards hostage so I had to choose something). So, again, to my mind this is not constructive or supportive of the "legitimacy" of the vote. I like the idea of having a static "Abstain" Rep that people can **opt into** once and forget about, particularly for large organizations like CEXes, etc.

Presuming that votes will occur on-chain and via a publicly auditable and transparent process, we lose the need to have a "strong voter turnout" showing in order to confirm validity of the vote because voter fraud, tampering, or manipulation of the results is a non-issue due to voting on blockchain. Having a strong voter numerical turnout is generally held up as one of the benefits of "compulsory voting" but within systems that do not have the same presumed protections against traditional fraud and manipulation.

Reply:
I see a lot of people calling this "compulsory voting". This is **absolutely not** compulsory voting. This is an acknowledgement that the voting system exists.

Reply:
Can't this acknowledgement just be buried by wallet providers? To redeem rewards the wallet provider can just default their users choice to abstain.

Reply:
indeed it could. and perhaps this is an excellent argument against it, because if no one even knows they've done this (even thought there is an audit trail), then we are back to worrying about the legitimacy of a vote.

Reply:
Here's an alternative: What if we, either through a ledger rule or best practice recommendation, required/suggested delegation at the same time as first delegation?

On the ledger side this could be: require the stake-address registration certificate to pick an initial dRep (including abstain).  Alternatively (or in addition) on the wallet side this becomes a simple extra step in the wallet onboarding wizard flow.

It has the "downside" (or upside, depending on your persuasion) that it wouldn't impact existing wallets, but it would be more aligned with trying to capture a voter when they're interested.  That is, they're already going through the effort of selecting an SPO, a gentle prompt to also learn about dReps if they wish could go a long way.

Reply:
I think this makes sense from a "UX" perspective. *"Hey, you just selected your first stake pool, that's awesome! Would you also like to browse available representatives and options for on-chain governance?"* I suppose the problem w/ this approach is that... do we have separate "certificates" for stake pool and "voter" delegation (so that they can be changed independently later)?

Reply:
yes, my reading of the proposal is that they are two separate certificates

Reply:
Side-thought that I just had thanks to your prompt @Quantumplation... Is there a punitive (negative) incentive for dReps who do not cast votes during a governance action? If we are going to apply negative incentive globally to all ecosystem participants and "force" (I use this word liberally) them to select a representative... and particularly if "representatives" are themselves positively incentivized (rewarded) by the system... will/should there be a negative penalty for failing to engage (as a representative). This idea largely springs from the idea of a user who has enough ADA and is so inclined to "Be their own representative" because attempting to use this "blocked rewards" as an incentive to participation is invalidated if there is something as simple as a 2A deposit to simply register yourself as a self-representative and never actually take any voting action.

Reply:
> two separate certificates

I actually want to make a new certificate that "can do it all": register your stake cred (if need be), delegate to a pool (if wanted), delegate to a drep (if wanted). and eventually maybe deprecate the old ones (though this is perhaps a nightmare wrt Plutus V1 and V2 script contexts). the plan wasn't to _require_ DRep registration, maybe that's a good idea, but even so it doesn't do anything for all the existing stake.

> Is there a punitive (negative) incentive for dReps who do not cast votes during a governance action?

No, and we don't necessarily want to do this. One way to achieve a lower governance action deposit (which I am personally hopeful for) is to accept the possibility of spam, and let off-chain socialization be the catalyst for how folks decide what to vote on.

Reply:
> No, and we don't necessarily want to do this. One way to achieve a lower governance action deposit (which I am personally hopeful for) is to accept the possibility of spam, and let off-chain socialization be the catalyst for how folks decide what to vote on.

Isn't this secretly kinda the same rationale and justification that can be applied to not **requiring** a forced on-chain voter registration for people? i.e. People must choose a representative (even if Abstain/No Contest) but there's no formal requirement (aside from social, off-chain) for their representative to live up to the delegation?

Reply:
Note that if pool selection, dRep selection, and stake registration all occur w/i the same certificate anyway, I view this as pretty much a non-issue. As then "abstain" simply becomes the default value for the "Vote Delegation" field wrt this "unified" on-chain certificate and is functionally identical to treating all the current registrations as abstained until they submit a new registration.

Reply:
> Isn't this secretly kinda the same rationale and justification that can be applied to not **requiring** a forced on-chain voter registration for people? i.e. People must choose a representative (even if Abstain/No Contest) but there's no formal requirement (aside from social, off-chain) for their representative to live up to the delegation?

I don't see the connection, sorry. The worry with a low registration is that you could get a high percentage of "yes" votes relative to the number of registered Lovelace, but if that denominator is too low it's not really a meaningful result. As for a DRep who is asleep at the wheel, you'd have a clear audit trail of that and can change your DRep as you see fit. In other words, picking your DRep is an off-chain issue, but that doesn't relieve the worry that 9 out of 10 Lovelace voted to set `k` to 5 (for example).

Reply:
Isn't this then more a matter of setting reasonable minimum quorum limits? As written, "abstained" stake would not count anyway. If we have 10,000,000 "registered" ADA, but 9,999,990 is "abstain" and 9 votes to move k to 5 and 1 votes **No Confidence** then that probably shouldn't be a resolution that can pass anyway. I would think we need to see a minimum **active** voting percentage as a share of circulating supply in order to enact a governance vote.

My large objection here is that "counting registered abstain as participating and conceding to the majority" works if all registrations are opt-in. When we remove that choice and use a (to the lay person w/o a deep technical understanding of blockchain) completely disconnected mechanic as the means of exerting pressure, it feels as if the choice to participate is removed. Therefore there ceases to be an option to "opt in".

Reply:
> As written, "abstained" stake would not count anyway. If we have 10,000,000 "registered" ADA, but 9,999,990 is "abstain" and 9 votes to move k to 5 and 1 votes **No Confidence** then that probably shouldn't be a resolution that can pass anyway. I would think we need to see a minimum **active** voting percentage as a share of circulating supply in order to enact a governance vote.

What you are describing is close to how the CIP was written before the update, which came with its own problems and which we pivoted away from (there's a section in the rationale about why we changed it, but it's very brief). We can certainly talk more about the pitfalls there.

---

> My large objection here is that "counting registered abstain as participating and conceding to the majority" works if all registrations are opt-in.

I would clarify this a bit, and say that it works if all registrations are _deliberate_.  I don't think that using language like "the choice to participate is removed" is fair, since delegating to a DRep is a choice to participate less, especially since anyone can delegate to `Abstain` or `No Confidence`.

Reply:
Maybe I should refine that to say, as written and to the lay-person understanding, it feels like the choice to participate in **consensus** via staking is removed if I choose not to participate in the socio-political governance layer. However, don't want to get lost in those semantics here. As stated above, if we can get to the "universal registration" certificate and the "default" state is simply "abstain" then I believe that is an elegant solution that works for all parties.

Reply:
Brief napkin thoughts on actual "actionable" revisions to the CIP as written:

- Remove/rename section title and content as written.
- Introduce new CDDL spec for stake delegation certificate to include voter representative delegation with default value of "abstain" that will be retroactively applied/considered default on all existing stake delegation certificates prior to the introduction of voter representatives.

**CDDL Spec**

```json
[
  2,
  [
    0,
    "addr_keyhash"
  ],
  [
    "0|1|2",
    "abstain (default)|no confidence|rep_keyhash"
  ]
]
```

I believe this accomplishes the same stated goals without the language and negative optics of "punitive incentives" and therefore is a more palatable solution.

Reply:
I don't think that applying "abstain" without people's consent accomplishes anything.

> choice to participate in consensus via staking is removed if I choose not to participate in the socio-political governance layer

I'll stop the back and forth soon here, I promise, I think we are going in circles. :) But I still find this language very unfair. Everyone continues to earn rewards and participate in block production with _all_ of their stake. Moreover, they absolutely do not have to participate in the governance layer, they only have to acknowledge that it exists (and abstain if they want) the next time that they want to use their reward account.

To be clear, this is by no mean the most important feature to me (personally), I am just trying to "steel man" it since all of the hyperbolic language that I am seeing makes me wonder if it is being misunderstood. As Andre said above, "Documents like this are all about making compromises." What _was_ important to me getting rid of the old AVST mechanics, and the compromise to add this negative incentive seemed fair enough to me to at least suggest it in the updated CIP.

Reply:
Yes, it boils down to the impasse of "Civic Right to Vote" vs. "Civic Duty to Vote" and the perceptions of individuals coming from one camp or another. IMO "Civic Right" side, which I find myself on, believes that everyone should **be able** to vote (including active abstention) but not be incentivized to vote (positively or negatively) as this makes voting a coerced act rather than the exercising of your rights being its own reward.

The "Civic Duty" camp believes "it's no big deal, hardly an inconvenience, if you don't like it, you can submit a blank vote, what's the big deal?" and are okay with compulsory incentives because of the moral imperative as a "citizen of the State".

Both have merits to their arguments and the impasse largely boils down to the "perception" of the thing rather than the "mechanics" of the thing.

Again, these arguments are simply my opinion and feelings as a member of the "Civic Right" camp, but IMO when we introduce the punitive measure (as minimal and light and hardly an inconvenience as it might be) then I no longer have a right and instead must behave in a certain way because "the system" is forcing me to (thus the friction).

My personal feeling at the moment is this compelled registration is going to be used to "force" a large portion of stake to choose "abstain" so that we can justify taking action even with minimal participation and that is why I feel it is "coercive" rather than "constructive" to the process. Note that treating inert stake as "abstained by default" is mechanically identical, only the optics and "talking points" have changed.

Also happy though to let this one go because I do believe it's an impasse and ultimately some compromise will be required for ratification of this or any other governance document and we'll never please 100% of people.

Reply:
> My personal feeling at the moment is this compelled registration is going to be used to "force" a large portion of stake to choose "abstain" so that we can justify taking action even with minimal participation and that is why I feel it is "coercive" rather than "constructive" to the process. Note that treating inert stake as "abstained by default" is mechanically identical, only the optics and "talking points" have changed.

I think this take is correct, users just checking a box ('Abstain') isn't meaningful participation. I think the same thing is achieved by just defaulting the choice to abstain.

Reply:
The difference between passive abstain and active abstain is that it demonstrates the person was given meaningful opportunity to consent to and participate in the new governance.  It's like checking an updated EULA; you get positive confirmation that some threshold of users are aware enough that *they could if they wanted to* (before anyone loses their mind, this is just an analogy, we're not asking people to sign an EULA.)

It provides a triggering threshold for exiting the bootstrapping era sooner than the timeout: "We know, via on-chain data, that at least this many people have seen the new wallet dialogs; if they *wanted* to be involved by this point, they could be, so it's now their own fault if we take the training wheels off and governance makes a decision they disagree with".

Note: I don't care how we acquire this signal; it doesn't need to be attached to a positive or negative incentive; I'm simply arguing for why the signal is valuable / meaningful.

Without this, we either prolong the bootstrap phase unnecessarily, or we have no bootstrap phase and decisions get made early on in governance that a majority of the stake *didn't meaningfully have the opportunity/notice to consent to*.

Reply:
The above comment stirred up an idea:
 - If this is about giving us signal to terminate the bootstrap phase early,
 - and the bootstrap phase already has an upper limit if we *don't* get the kind of participation we want
 - the reward gate could also expire on the same timeline.

Reply:
Can't we still accomplish (via the universal certificate method) the same monitorable, on-chain "signal" that we have a sufficient quorum to proceed to exit bootstrap phase without the reward gate?

As proposed, wallets will have to build out a "Governance UX" either way. Submitting an updated "delegation" certificate is an identical process (only one additional object in the certificate) to existing systems. In the same way we had "on chain indicators" for readiness to complete the Babbage HF we can promote via social pressure "Get us out of bootstrap! Go to your wallet and select a representative option today!" to encourage participation.

Finally, we remove any language about ever touching or messing w/ people's rewards while accomplishing all of the stated goals. New wallets from the time that this new Delegation Certificate is implemented will always carry "implicit acknowledgement of governance" so, again, we don't need the threat of the stick to compel action.

Reply:
That's what I was trying to address above:

> I suppose that the bootstrap together with a mandatory DRep decision with stake pool delegation would achieve the same thing, but we don't have time to gracefully deprecate the existing certificates.

Reply:
Ah gotcha, I misunderstood "deprecating the existing certificates" :)